packages/wdio-utils/src/shim.ts

```diff
@@ -119,11 +119,16 @@ let executeHooksWithArgs = async function executeHooksWithArgsShim<T> (hookName:
         args = [args]
     }
 
+    /**
+     * save `this` pointing to caller browser object
+     */
+    const thisBrowser = this;
```

Comment:
> Can we add a smoke test to ensure we have the `browser` object available in all hooks?

Ehm, probably we can though I still can't properly setup the package because of error; moreover, the next command (npm build) seems to require compiling some stuff that lasted forever so I interrupted it.

> I am not aware that we execute `executeHooksWithArgs` with the browser object scope. E.g. [here](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-cucumber-framework/src/index.ts#L268-L272) where we run it as following:

Well, I was basing on `wrapCommandFn` that calls `executeHooksWithArgs` with `this` assigned which is actually the call chain in my scenario; I wasn't aware of another places calling it directly. Maybe we could make `this` assigned only in this case for the beginning and then think where else it makes sense? I can't determine the most acceptable option b/c I just saw the sources and they're quite big and complicated.

Reply:
> I still can't properly setup the package because of error

I fixed the build, please rebase on top of main.

> the next command (npm build) seems to require compiling some stuff that lasted forever so I interrupted it.

Can you explain what it requires to compile? As described in our contributing guidelines, you should run `npm run setup-full` and then run `npm run watch` for development.

> Maybe we could make `this` assigned only in this case for the beginning and then think where else it makes sense?

I would love to make this change everywhere so that the behavior is consistent.

> I can't determine the most acceptable option b/c I just saw the sources and they're quite big and complicated.

If you search for `executeHooksWithArgs` in the codebase you will find a handful places where the function is being used. We should update these places. I am happy to help navigate through the code base. If you need support feel free to book a slot as part of our [office hours](https://webdriver.io/community/openofficehours).

Reply:
> I fixed the build, please rebase on top of main.

That fix is not related, the reason is `    "postinstall:vulnerabilities": "cat ./.husky/postinstall.txt"` line in `package.json`. Windows has no `cat` command. Removed it and installation succeeded. Making the next step...

> I would love to make this change everywhere so that the behavior is consistent.

Ok, I got it but what about tests? The function is called directly when there's no browser available. I guess `null` is allowable in this case.

So the idea is to change all `executeHooksWithArgs(...)` calls to `executeHooksWithArgs.call(browser, ...)`?

Reply:
> Can you explain what it requires to compile? As described in our contributing guidelines, you should run `npm run setup-full` and then run `npm run watch` for development.

I run `npm run setup-full`, it does something for a while and then fails on building `fibers`. This step involves MSBuild and MSVC installed on my machine so the issue is somewhere in them but I had no ability yet to dig deeper.

Reply:
> So the idea is to change all `executeHooksWithArgs(...)` calls to `executeHooksWithArgs.call(browser, ...)`?

That seems reasonable to me.

> it does something for a while and then fails on building `fibers`

Are you running Node.js v16?

Reply:
> Are you running Node.js v16?

Nope. They dropped W7 which I'm currently on so I use v12. Does it make sense? I'll try the last W7-compatible v14 release but all that requires too much hassle behind one tiny change :(

Reply:
We recommend to use Node.js v16 for developing on WebdriverIO. I think the change here is not trivial. I would like to have a consistent behavior across all WebdriverIO hooks. So it would be great if we could adapt other parts of the code base as well.

Reply:
Currently I succeeded running v16 (seems they returned W7 support) but `npm run setup-full` fails on some stage with ECONNRESET. Looks like it's unstable network connection that's the reason (though failing the whole huge process just because of a single temporary disconnect is not what NPM could boast of). I'll try to find more stable connection and clean OS for install